### ISSUE TITLE
[HOLD for payment 2024-10-30] De-dupe OpenReport requests

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#51072]
similarly to this issue for OpenApp https://github.com/Expensify/App/issues/50076

### Problem
We are seeing some scenarios where multiple OpenReport requests are serialized in a row. When multiple such requests are sent, it's redundant, slows down the app, and contributes to server traffic.

### Solution
De-dupe OpenReport requests for **the same reportId** such that only one can be serialized at a time.

<details><summary>Issue Owner</summary>Current Issue Owner: @strepanier03</details>

**Comment by gedu** [comment]
Hey, I'm Edu from Callstack, I can work on it

**Comment by eh2077** [comment]
@mountiny Can I get assigned here? So we can track the payment, thanks!

**Comment by mountiny** [comment]
@eh2077 can you please complete the checklist? I think we do not need regression tests for this as we added unit tests

**Comment by eh2077** [comment]
### Checklist

- [x] [@eh2077] The PR that introduced the bug has been identified. Link to the PR: _**This is an improvement, so there isn't a PR causes this bug.**_

TL;DR
Digging into the commit history, we have pursued this improvement for some time. Below are the timeline
1. A de-dupe logic was first introduced to solve this issue https://github.com/Expensify/App/pull/39024
2. The de-dupe logic was reverted https://github.com/Expensify/App/pull/40180 to pursue better solution
3. Then https://github.com/Expensify/App/pull/47913 was merged 
4. This PR can be considered as a follow-up PR of https://github.com/Expensify/App/pull/47913


- [x] [@eh2077] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A
- [x] [@eh2077] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A
- [x] [@eh2077] Determine if we should create a regression test for this bug. **_No, because this improvement is implicit and doesn't impact the flow to use the App._**

**Comment by strepanier03** [comment]
@eh2077 - feel free to request payment now. Cheers!

**Comment by eh2077** [comment]
Requested in NewDot

**Comment by JmillsExpensify** [comment]
$250 approved for @eh2077
